### PR TITLE
Allow editing a request without a redirect change

### PR DESCRIPTION
**Pull request by kevindew**
This resolves an issue where users could not edit a request once the
request was accepted as the validator thought a redirect already
existed.

This has been done to resolve this 2nd link zendesk ticket: https://govuk.zendesk.com/agent/tickets/3806774

It also includes a couple of fixes for things I found while running the app for the first time.

### CODE REVIEW

**Comment by cbaines** [comment]
> This resolves an issue where users could not edit a request once the
> request was accepted as the validator thought a redirect already
> existed.

What kind of edit's should be allowed once the request has been accepted? I guess some edit's a rejected as they change what has been reviewed.

**Comment by kevindew** [comment]
> > This resolves an issue where users could not edit a request once the
> > request was accepted as the validator thought a redirect already
> > existed.
> 
> What kind of edit's should be allowed once the request has been accepted? I guess some edit's a rejected as they change what has been reviewed.

The situation in the zendesk ticket is wanting to change the organisation that owns the request.

I also question the logic in editing a request since that is something that happened at a particular time by a particular person. But it seems that the request and redirect are tightly coupled and once accepted you affectively treat the request as the current status of the redirect rather than as an artefact of time.

File: app/models/short_url_request.rb [hunk 1]
```diff
@@ -63,11 +63,12 @@ def uses_advanced_options?
...
-           to_path: to_path,
-           route_type: route_type,
-           segments_mode: segments_mode,
-           override_existing: override_existing,
+           :from_path => from_path,
+           :to_path => to_path,
+           :route_type => route_type,
+           :segments_mode => segments_mode,
+           :override_existing => override_existing,
+           :short_url_request.ne => self,
```
[the remark this answers is not in the document]
Comment (kevindew) [reply app/models/short_url_request.rb]: it means not equals. Mongoid took the unusual route of applying methods to symbols to specify operators.

This condition is used to allow validation to pass if a redirect already exists with the attributes of the request so long as that redirect is the one associated with that request.

File: config/environments/development.rb [hunk 2]
```diff
@@ -51,4 +51,9 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   # config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+
+  # Tell Action Mailer not to deliver emails to the real world.
+  # The :file delivery method writes emails that would be sent to the
+  # application log file
+  config.action_mailer.delivery_method = :file
```
Comment (DilwoarH) [inline config/environments/development.rb]: Will this be an issue? ie. will the logs take up too much disk space?

File: app/models/short_url_request.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbaines) [reply app/models/short_url_request.rb]: Right, ok, I forgot the Short URL Manager uses MongoDB.

**Comment by kevindew** [comment]
Thanks @cbaines

**Pull request merged**